tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +56,43 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws error when checking the directory if lacks writing permissions',
+    () => withTempDir(
+    (tmpDir) => {
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      return fs.mkdir(tmpPath, '0622').then(() => {
```

Comment:
I prefer to see tests with real files using the temp dir (like the existing ones). However, in cases like lacking write-permissions it might be too hard to set up a test for that with real files. I think it would be fine to use a mock for that but only as a last resort.